sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1229,3 +1229,98 @@ case class Flatten(child: Expression) extends UnaryExpression {
 
   override def prettyName: String = "flatten"
 }
+
+/**
+ * Returns the array containing the given input value (left) count (right) times.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(element, count) - Returns the array containing element count times.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_('123', 2);
+       ['123', '123']
+  """)
+case class ArrayRepeat(left: Expression, right: Expression)
+  extends BinaryExpression {
```

Comment:
What about simplifying the type checking with the below snippet? 
```
with ExpectsInputTypes {
 override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType, IntegerType)
...
```

Reply:
Good point, changed in commit 6e699e0